corehq/apps/app_manager/suite_xml/sections/details.py

```diff
@@ -189,9 +189,11 @@ def build_detail(self, module, detail_type, detail, detail_column_infos, tabs=No
             # Add actions
             if detail_type.endswith('short') and not module.put_in_root:
                 if module.case_list_form.form_id:
-                    target_form = self.app.get_form(module.case_list_form.form_id)
-                    if target_form.is_registration_form(module.case_type):
-                        d.actions.append(self._get_reg_form_action(module))
+                    from corehq.apps.app_manager.views.modules import get_parent_select_followup_forms
+                    form = self.app.get_form(module.case_list_form.form_id)
+                    valid_forms = [f.unique_id for f in get_parent_select_followup_forms(self.app, module)]
```

Comment:
Does this need to check against `valid_forms`, given that the app build validation would have checked?

Reply:
I have mainly added, because right now, it's checking for `is_registration_form` here as well. To be complete, we would need to check whether it's a valid followup form. Generally though, I think it's good to check here.